contracts/test/CW721toERC721PointerTest.js

```diff
@@ -0,0 +1,135 @@
+            expect(result).to.deep.equal({data:{
+                approvals:[
+                    {spender: accounts[1].seiAddress, expires:{never:{}}}
+                ]}});
+        });
+
+        it("should verify if an operator is approved for all tokens of an owner", async function () {
+            const result = await queryWasm(pointer, "operator", { owner: accounts[0].seiAddress, operator: admin.seiAddress });
+            expect(result).to.deep.equal({
+                data: {
+                    approval: {
+                        spender: admin.seiAddress,
+                        expires: {never:{}}
+                    }
+                }
+            });
+        });
+
+        it("should retrieve contract information", async function () {
+            const result = await queryWasm(pointer, "contract_info", {});
+            expect(result).to.deep.equal({data:{name:"MyNFT",symbol:"MYNFT"}});
+        });
+
+        it("should fetch NFT info based on token ID", async function () {
+            const result = await queryWasm(pointer, "nft_info", { token_id: "1" });
+            expect(result).to.deep.equal({ data: { token_uri: 'https://sei.io/token/1', extension: '' } });
+        });
+
+        it("should fetch all information about an NFT", async function () {
+            const result = await queryWasm(pointer, "all_nft_info", { token_id: "1" });
+            expect(result).to.deep.equal({
+                data: {
+                    access: {
+                        owner: accounts[0].seiAddress,
+                        approvals: [
+                            {
+                                spender: accounts[1].seiAddress,
+                                expires: {
+                                    never: {}
+                                }
+                            }
+                        ]
+                    },
+                    info: {
+                        token_uri: "https://sei.io/token/1",
+                        extension: ""
+                    }
+                }
+            });
+        });
+
+    })
+
+    describe("execute operations", function () {
+        it("should transfer an NFT to another address", async function () {
+            await executeWasm(pointer, { transfer_nft: { recipient: accounts[1].seiAddress, token_id: "3" }});
+            const ownerResult = await queryWasm(pointer, "owner_of", { token_id: "3" });
+            expect(ownerResult).to.deep.equal({ data: { owner: accounts[1].seiAddress, approvals: [] } });
+            await (await erc721.connect(accounts[1].signer).transferFrom(accounts[1].evmAddress, accounts[0].evmAddress, 3)).wait();
+            const ownerResult2 = await queryWasm(pointer, "owner_of", { token_id: "3" });
+            expect(ownerResult2).to.deep.equal({ data: { owner: accounts[0].seiAddress, approvals: [] } });
+        });
+
+        it("should approve a spender for a specific token", async function () {
```

Comment:
would be good to test that the spender can only spend the token after given approval and can no longer after approval has been revoked